agent/src/main/java/com/teamscale/jacoco/agent/upload/teamscale/TeamscaleUploader.java

```diff
@@ -33,21 +41,51 @@ public void upload(CoverageFile coverageFile) {
 			if (tryUploading(coverageFile)) {
 				deleteCoverageFile(coverageFile);
 			} else {
-				logger.warn("Failed to upload coverage to Teamscale. " +
-								"Won't delete local file {} so you can upload it yourself manually.",
-						coverageFile);
+				logger.warn("Failed to upload coverage to Teamscale. "
```

Comment:
Was it a conscious decision to only implement the re-try the upload to Teamscale? The upload to artifactory could fail in a similar way. In case it was left out, please document and also add the why.